include/networkit/centrality/GroupClosenessGrowShrink.hpp

```diff
@@ -0,0 +1,665 @@
+/*
```

Comment:
Whats the rationale behind implementing `GrowClosenessGrowShrink` and `Dijkstra` as header only?

Reply:
Because they are templated, moving the implementation to a source file would yield compile-time errors.

Reply:
A templated Dijskstra is reasonable but why do we need a templated Grow-Shrink?

Reply:
To handle weighted/unweighted graphs. I wanted to specialize the code for the two version to avoid doing many run-time checks for `G.isWeighted()` .

Reply:
As far as I can see, this does not remove the need to perform runtime checks, it only specializes the code for `count` weights instead of `double` weights. I'm not sure if that is really worthwhile (double math should only be slightly slower than integer math). If you deem it worthwhile, use the pimpl idiom to avoid putting the entire algorithm into the header.